pkg/kubelet/kuberuntime/kuberuntime_container_linux.go

```diff
@@ -61,28 +61,33 @@ func (m *kubeGenericRuntimeManager) generateLinuxContainerConfig(container *v1.C
 		lc.SecurityContext.NamespaceOptions.TargetId = nsTarget.ID
 	}
 
+	return lc
+}
+
+// generateLinuxContainerResources generates linux container resources config for runtime
+func (m *kubeGenericRuntimeManager) generateLinuxContainerResources(pod *v1.Pod, container *v1.Container, enforceMemoryQoS bool) *runtimeapi.LinuxContainerResources {
 	// set linux container resources
-	lc.Resources = m.calculateLinuxResources(container.Resources.Requests.Cpu(), container.Resources.Limits.Cpu(), container.Resources.Limits.Memory())
+	lcr := m.calculateLinuxResources(container.Resources.Requests.Cpu(), container.Resources.Limits.Cpu(), container.Resources.Limits.Memory())
```

Comment:
Hello, I think of a case.

After the container resource is resized, the resize state is Deferred or Infeasible. At this time, when the container image is updated, the resource passed to containerd is wrong, because the resource cannot be rolled back, and the container will fail to start at this time.

For example, when the resize status is Deferred, after updating the container image, start the container and report an error:
```
failed to create containerd task: OCI runtime create failed: container_linux.go:380:
          starting container process caused: process_linux.go:545: container init
          caused: process_linux.go:508: setting cgroup config for procHooks process
          caused: failed to write "8000000": write /sys/fs/cgroup/cpu,cpuacct/kubepods/burstable/pod65ead451-b6bf-407b-a90d-ea2b193384cf/4abf6eb7acfc2d43cb69b866f1fe47ac4ef4593215dfbfc575341149367f7dce/cpu.cfs_quota_us:
          invalid argument: unknown'
        reason: StartError
```

Reply:
@zuoshaoyi Thanks for another great catch. I have created issue https://github.com/kubernetes/kubernetes/issues/111592 to track fixing this.

I have to look closely, but I think the right fix would be to use the values in resourcesAllocated when restarting containers if awaiting resize.